### DISCUSSION TITLE
Preheating issues and have to power printer on and off

### DISCUSSION THREAD

**Discussion by cainedk** [Jyers/Marlin#768] [General Discussion]
Hi
I Just flashed E3V2-BLTouch-3x3-v4.2.2.bin to my Ender 3 v2 today.
But I am having some issues where then I start a preheat the printer will start to heat up but at some time the printer will start to beep and display an message where it says "Print killed" and something about heating issues.

Before flashing this firmware I have not had any issues with the temperature on my printer and when I flash the stock firmware from Creality to my printer there are no problems with the temperature as well,.
When I flash the firmware E3V2-BLTouch-3x3-v4.2.2.bin the printer will then again the temperature error again.

I have been able to get it to show when starting preheat and while its heating I set the printer to home.
Have also been able to get the issue when navigating to other menus while preheating.
But its not something that I with have been able to reproduce 100% success.

YES I have looking my printer over and there are no faulty hardware.

Will try and remember to snap a photo of the error next time I get it.

**Comment by Cal-LF** [comment]
I get the same thing when preheating after printing. Have to restart printer but will sometimes repeat this unless it has cooled for a while. Changes were made in the thermal protection setting recently and may be the cause.
This has happen on my last print and the one before.

**Comment by Jyers** [comment]
I would recommend running a PID autotune to make sure your heater is working optimally. I recently increased the safety of the thermal runaway values as they were a bit conservative, but I may have gone a bit too far. I'll look into changing them slightly again to reduce the frequency of such errors.

**Reply by Cal-LF** [reply]
Where are those settings. I may try different values too.
After doing a bed PID tune, the values are very far off from stock and the temp swings much more. Not sure I trust PID auto tune.

**Reply by Tuncay-Ayhan** [reply]
> Where are those settings. I may try different values too.
> After doing a bed PID tune, the values are very far off from stock and the temp swings much more. Not sure I trust PID auto tune.

The PID settings can vary per location. Even moving your printer from one room to another could possibly mean that the values change again.

**Reply by kielerkoenig** [reply]
Please do set the runaway values back to the previous settings. 
I did a PID autotune after also running into the "Thermal Runaway" issue above and finding this thread here.
Did actually make things worse, printer compensated way too far before reaching target temperature and I got "Heater Failure" error instead. 
Changed back to original PID values and flashed 1.3.1 instead where the runaway settings are (hopefully) more conservative. 
Loved the extended menus though, so would love to return to 1.3.2 if this issue is fixed.

Thanks for the beautiful UI by the way, you provide what Creality is lacking ;)
Edit: Ender 3V2 4.2.2 Board, genuine BL-Touch, stock hotend

**Reply by Jyers** [reply]
I'm not going to put them all the way back as with those settings there was a report of someone who has their hotend go up in smoke without thermal runaway tripping... I will however try to find a happy medium.

**Comment by Cal-LF** [comment]
Changed time out settings for thermal protection back to V1.3.1 and have no thermal alert so far. Also found the new "purge extra filament" feature when using Cura change filament plugin, which now works. 
I think the WATCH_TEMP_INCREASE 2   // Degrees Celsius is the culprit causing the temp alert, it was set at 4.
Config_adv.h
 #define THERMAL_PROTECTION_PERIOD 40  // Seconds
 #define WATCH_TEMP_PERIOD 40   // Seconds
 #define WATCH_TEMP_INCREASE 2   // Degrees Celsius
 #define THERMAL_PROTECTION_BED_PERIOD  180 // Seconds
 #define WATCH_BED_TEMP_PERIOD  180 // Seconds

**Reply by Jyers** [reply]
@Cal-LF If you're already testing it, would you mind trying it at 3? As I said above, I'm aiming at safety here, but I don't want everyone having errors. So if you find it works fine at 3 I want to go for as high as I can get it without too many errors.

**Reply by Cal-LF** [reply]
Yes, for sure. I will try decimal also. 2.5, 3.5

**Reply by Cal-LF** [reply]
The 2.5 and 3.5 deg do not work, get a failed compile. The 3 deg works.
New settings:
#define THERMAL_PROTECTION_PERIOD 30 // Seconds
#define WATCH_TEMP_PERIOD 30 // Seconds
#define WATCH_TEMP_INCREASE 3 // Degrees Celsius
#define THERMAL_PROTECTION_BED_PERIOD 120 // Seconds
#define WATCH_BED_TEMP_PERIOD 120 // Seconds

**Reply by Jyers** [reply]
3 works as in it compiles, or it doesn't trigger thermal errors?

**Reply by cainedk** [reply]
Please include link instead of a screenshot

**Comment by cainedk** [comment]
As my printer is almost stock other than the Bltouch and is running on the 4.2.2 board I just ran the auto PID for the hotend and the bed and have not had any issues with the heating error since I flashed the 1.3.2 firmware
Again this was just something that first showed when I went from stock firmware to this firmware 
But now its running good ;)

**Reply by Cal-LF** [reply]
This was happening to me only after moving to the latest release. No problems now with the above settings.

**Reply by roliam56** [reply]
I confirm, it's ok

**Reply by Cal-LF** [reply]
@Jyers Your question about the 3 working. Answer is yes to both. 3 compiles and does not trigger thermal errors.

**Reply by Jyers** [reply]
Great, I'll leave that the default for now and keep an eye out if it needs any more tweaking
